utils/updateTestPlanReadme.pl

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+
+use Pod::Usage;
+use Getopt::Long;
+use File::Basename;
+
+# command line options
+my $dirloc = '.'; # directory for serching test cases
+my $help;
+my $DEBUG = 0;
+
+# global variables
+my $tcCounter = 0;
+
+sub main {
+    GetOptions(
+        'help|?'  => \$help,
+        'dir|d=s' => \$dirloc,
+        'debug'   => \$DEBUG,
+    ) or pod2usage(2);
+
+    if ($help) {
+        pod2usage(1);
+        exit;
+    }
+    opendir(my $dir, $dirloc) || die "cannot open directory $dirloc";
+
+    my @files = readdir $dir;
+    @files = sort {$a cmp $b} @files;
+    foreach my $f (@files) {
+        print "$dirloc/$f\n" if $DEBUG;
+        next if $f =~ /^\./;
+
+        if ( -d "$dirloc/$f" and "$f" =~ /-TP/ ) {
+            tcList( "$dirloc/$f" );
+        }
+    }
+    print "No test cases found\nUse -d to specify directory\n" if !$tcCounter;
+}
+
+sub tcList
+{
+    my $tcDir = shift;
+
+    my $output = "## Test cases list\n\n";
+    $output .= "<!-- Table generated by script " . basename ($0) . " from Zonemaster/Zonemaster utils directory -->\n\n";
+    $output .= '|Test Case |Test Case Description|' . "\n";
+    $output .= '|:---------|:--------------------|' . "\n";
```

Comment:
Since the first line is also generated by the script, I think the comments should be before:
```
    my $output.= "<!-- Content until EOF generated by script " . basename ($0) . " from Zonemaster/Zonemaster utils directory -->\n\n";
    $output .= "## Test cases list\n\n";
    $output .= '|Test Case |Test Case Description|' . "\n";
    $output .= '|:---------|:--------------------|' . "\n";
```

Else your model could be used for generateTestCaseList.pl too to make simpler to run.

Reply:
Done.